runtime/grid_reduction.cu

```diff
@@ -857,6 +878,30 @@ __device__ void iterGroupedGridReduce(
         index_utils::maskedOffset<!X_THREAD, !Y_THREAD, !Z_THREAD>(
             threadIdx, blockDim);
 
+    // Max vectorized load/store size is 16 bytes, if each thread has more than
+    // 16 bytes, split into multiple sections to ensure each thread occupies
+    // only 16 bytes at most. For example, if each thread has 8 fp32 which
+    // occupies 32 bytes, split into 2 sections, in each secdtion each thread
+    // holds 4 fp32 or 16 bytes. Thread-0 processes elements [0,7], the first 4
+    // elements [0,3] are stored in the first section and the last 4 elements
+    // [4,7] are stored in the 2nd section. The data layout in gmem is:
+    //         |-----------section 1-----------|-----------section 2-----------|
```

Comment:
IIUC, section 1 is the first load and section 2 is the second.
`gmem_offset_inter` is the stride for each load (length of a section).
`gmem_offset_intra` is the offset to the beginning of the section.

Reply:
It's not a comment just trying to verify I was reading the code correctly.

Reply:
yes, you are reading the code correctly. I renamed these two variables to more explicitly express their function.
`gmem_offset_inter` --> `elements_per_section`
`gmem_offset_intra` --> `offset_in_section`